### DISCUSSION TITLE
Final report

### DISCUSSION THREAD

**Discussion by MFJI** [Xilinx/finn#1148] [Q&A] [answered]
I am using an alveo board in FINN. Where can I see the final report after implementation of resource utilization and power consumption from vivado after generate bit file? I can only see the estimate and post synth resource report in the output/report folder. Thank you.

**Comment by auphelia** [comment]
Hi @MFJI ,
You can open the project in Vivado to get results like power consumption.

**Reply by Ba1tu3han** [reply]
Hello @auphelia ,

Should it be used only xpr file to analyze results or any finn exported file else is required ?

**Reply by auphelia** [reply]
Hi @Ba1tu3han, you can use the finn exported files as well. But we only report resources and timing and don't report power. The reports are output products of the synthesize bitfile step: https://github.com/Xilinx/finn/blob/dev/src/finn/builder/build_dataflow_steps.py#L811 

In general, for the final results FINN extracts the information directly from the Vivado/Vitis reports. So, you can either use the FINN generated reports or if they don't contain the values you are looking for, you can use Vivado/Vitis.

**Reply by Ba1tu3han** [reply]
Thank you for your reply.

Should I import just xpr file to Vivado? or bit/hwh file is also neeed to analyze power consumption? 

It is really rookie question sorry for my limited experience.

**Reply by JPPalacios** [reply, marked as answer]
Hi @Ba1tu3han,

You can navigate to where the *.xpr file is (it will have all the other Vivado files like sim/src/etc.) and run `vivado <proj_name>.xpr` and you should see all the project info as soon as it opens. Be sure to source your Vivado tool if you're on Linux! `source /opt/Xilinx/Vivado/<YEAR>/settings64.sh`

**Reply by MFJI** [reply]
Hi @auphelia,

Thank you for your answer. I still have a question. Where can I find the project that I can have the reports in Vivado?

**Reply by JPPalacios** [reply]
Hi @MFJI,

I believe these are in your `FINN_HOST_BUILD_DIR` path (see [getting started docs](https://finn.readthedocs.io/en/latest/getting_started.html)) under the name `vivado_zynq_proj_*` directory. You should follow the steps mentioned above to open the Vivado project.

**Reply by MFJI** [reply]
Hi @JPPalacios,

Thank you for your answer. I am using an Alveo board, I can't see any  directory named vivado_zynq_proj_*, I think that is the one for pynq board. There are several directories in the FINN_HOST_BUILD_DIR path seem to be the project, but none of them is finished with implementation, so I don't know which is the directory I should look into.

**Reply by fpjentzsch** [reply]
If you sort by time you should see which directories FINN worked on last. For Alveo the final Vivado project of the accelerator should be called "vivado_stitch_proj_*", which is then linked with the shell and DMAs in a "vitis_link_proj_*".
